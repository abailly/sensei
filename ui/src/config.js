export const config = {
  user: USER
}
