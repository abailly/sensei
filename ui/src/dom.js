/**
 * @fileOverview
 * @name dom.js
 * @author arnaud@pankzsoft.com
 * @license MIT
 */



/**
 * Basic implementation of JSX element transformation function.
 * Stolen from <a href='https://blog.r0b.io/post/using-jsx-without-react/'>this blog post</a>.
 *
 * @param {String} tagName the name of the tag to create
 * @param {Object} attrs an object defining all the attributes of the new node
 * @param {any} children a possibly empty sequence of children to add to this node.
 * @returns {Object} a newly initialized DOM element with given attributes and children
 */
export function dom(tagName, attrs = {}, ...children) {
  const elem = Object.assign(document.createElement(tagName), attrs);
  for (const child of children) {
    if (Array.isArray(child)) elem.append(...child);
    else elem.append(child);
  }
  return elem;
}

export function clear(parentId) {
  const timelines = document.getElementById(parentId);
  clearElement(timelines);
}

export function clearElement(elem) {
  while (elem.firstChild) {
    elem.removeChild(elem.firstChild);
  }
}
