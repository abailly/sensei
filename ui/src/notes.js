import showdown from 'showdown';

function formatNote(note) {
  return "<div class='note'>" +
    new showdown.Converter({ simplifiedAutoLink: true }).makeHtml('#### ' + new Date(note.noteStart).toLocaleTimeString() + '\n\n' + note.noteContent) +
    "</div>";
}

/**
   Draw a timeline containing notes
*/
export function drawNotes(container, notesData) {
  const chart = new google.visualization.Timeline(container);
  const dataTable = new google.visualization.DataTable();

  dataTable.addColumn({ type: 'string', id: 'Role' });
  dataTable.addColumn({ type: 'string', id: 'dummy bar label' });
  dataTable.addColumn({ type: 'string', role: 'tooltip', p: { 'html': true } });
  dataTable.addColumn({ type: 'date', id: 'Start' });
  dataTable.addColumn({ type: 'date', id: 'End' });
  notesData.forEach(note => {
    let start = new Date(note.noteStart);
    dataTable.addRow(['Notes', '', formatNote(note), start, new Date(start.getTime() + 60000)]);
  });
  var options = {
    tooltip: { isHtml: true },
  };
  chart.draw(dataTable, options);
}
