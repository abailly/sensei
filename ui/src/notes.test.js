import {describe, expect, test} from '@jest/globals';
import {dom} from './dom.js';
import {config} from "./config";


jest.mock('./request.js', () => (
    {
        get: jest.fn((url, callback) => callback([{noteStart: '2020-10-12T14:30', noteContent: 'content'}], {}))
    }
))

import notes from './notes';
import Navigo from "navigo";

describe('Notes', () => {

    let router;
    let container;

    beforeEach(() => {
        router = new Navigo(null, false);
        container = document.createElement('div');
    });

    test('it should display note date like 2020-10-12 14:30', () => {
        notes.list(router, container, 1);
        expect(container).toMatchSnapshot();
    });
})