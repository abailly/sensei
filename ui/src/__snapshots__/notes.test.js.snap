// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notes it should display note date like "2020-10-12" 1`] = `
<div>
  <div
    class="pagination"
  >
    <button
      class="btn-prev"
      disabled=""
    >
      &lt;&lt;
    </button>
    <button
      class="btn-next"
      disabled=""
    >
      &gt;&gt;
    </button>
  </div>
  <div
    id="daily-notes"
  >
    <h2>
      Notes for 
      1
       
    </h2>
    <div
      class="note-full"
    >
      <h3>
        <a
          href="http://localhost/notes/2020-10-12"
        >
          2020-10-12 14:30
        </a>
      </h3>
      <h4>
        undefined
      </h4>
      <h4
        class="note-tags"
      />
      <div
        class="note-content"
      >
        <p>
          content
        </p>
      </div>
    </div>
  </div>
</div>
`;
